elephant/phase_analysis.py

```diff
@@ -307,3 +307,80 @@ def phase_difference(alpha, beta):
     delta = alpha - beta
     phase_diff = np.arctan2(np.sin(delta), np.cos(delta))
     return phase_diff
+
+
+def weighted_phase_lag_index(signal_i, signal_j, sampling_frequency,
+                             absolute_value=True):
+    r"""
+    Calculates the Weigthed Phase-Lag Index (WPLI)
+
+    This function expects the two signals (each containing multiple trials).
+    It Fourier-transforms the signals and for each trial pair, it calculates
+    the cross-spectrum. With that the WPLI is calculated.
+
+    Parameters
+    ----------
+    signal_i, signal_j: (t, n) np.array
+        Time-series of the first and second signals,
+        with `t` time points and `n` trials.
+    sampling_frequency: quantity/ scaler
+        Sampling frequency of the signals in Hz.
+    absolute_value: boolean (default: True)
+        Takes the absolute value of the numerator in the WPLI-formula.
+
+    Returns
+    -------
+    wpli: float
+        Weighted phase-lag index of signal_i and signal_j across trials.
+        Range: :math: `[0, 1]`
+
+    Notes
+    -----
+    This implementation is base on the formula taken from [1] (pp.1550):
+    .. math::
+        WPLI = \frac{| E( |Im(X)| * sgn(Im(X)) ) |}{E( |Im(X)| )}
+
+    with:
+    E{...} : expected value operator
+    Im{X} : imaginary component of the cross-spectrum
+    X = Z_1 * Z_2_conjugate : as cross-spectrum
+    X = R * exp(i * theta) : R = magnitude and theta = relative phase
+    Z = A * Y
+    where
+    A: real-valued matrix with frequency dependent coefficients
+    Y: complex-valued vector, representing the Fourier spectra
+    of a particular frequency
+
+    Usage/Interpretation of the parameter 'absolute_value':
+    When set to 'False', the WPLI contains additional directionality
+    information about which signal leads/lags the other signal:
+        - wpli > 0 : first signal leads second one
+        - wpli < 0 : first signal lags second one
+
+    References
+    ----------
+    [1] "An improved index of phase-synchronization for electrophysiological
+    data in the presence of volume-conduction, noise and sample-size bias"
+    by Martin Vinck, Robert Oostenveld, Marijn van Wingerden,
+    Franscesco Battaglia, Cyriel M.A. Pennartz
+    """
+    if np.shape(signal_i) != np.shape(signal_j):
+        raise ValueError("trial number and trial length of signal i and j "
+                         "must be equal")
+
+    # ARRAY-approach
+    # calculate Fourier transforms
+    fft1 = np.fft.rfft(signal_i)
+    fft2 = np.fft.rfft(signal_j)
+    freqs = np.fft.rfftfreq(np.shape(signal_i)[1], d=1.0 / sampling_frequency)
+
+    # obtain cross-spectrum
+    cs = fft1 * np.conjugate(fft2)
+    # calculate WPLI
+    wpli_num = np.mean(np.abs(np.imag(cs)) * np.sign(np.imag(cs)), axis=0)
+    if absolute_value:
+        wpli_num = np.abs(wpli_num)
+    wpli_den = np.mean(np.abs(np.imag(cs)), axis=0)
+    wpli = wpli_num / wpli_den
+
+    return freqs, wpli
```

Comment:
As such, the function returns the most interesting thing, `wpli`, second, whereas the first return `freqs` is the more uninteresting thing. I would strongly suggest ot flip this. Otherwise often you have to type
```
,wpli = weighted_phase_lag()...
```